packages/blade/src/components/Alert/Alert.tsx

```diff
@@ -0,0 +1,217 @@
+const intentIconMap = {
+  positive: AlertIcon,
+  negative: AlertIcon,
+  information: AlertIcon,
+  notice: AlertIcon,
+};
+
+const Alert = ({
+  description,
+  title,
+  isDismissable = true,
+  onDismiss,
+  contrast = 'low',
+  isFullWidth = false,
+  intent = 'information',
+  isBorderless = false,
+  actions,
+}: AlertProps): ReactElement | null => {
+  const [isVisible, setIsVisible] = useState(true);
+  const contrastType = contrast === 'high' ? 'highContrast' : 'lowContrast';
+
+  const Icon = intentIconMap[intent];
+  const icon = <Icon color={`feedback.icon.${intent}.${contrastType}`} size="large" />;
+
+  const _title = title ? (
+    <Box marginBottom="spacing.1">
+      <Heading variant="small" contrast={contrast}>
+        {title}
+      </Heading>
+    </Box>
+  ) : null;
+
+  const _description = <Text contrast={contrast}>{description}</Text>;
+
+  const primaryAction = actions?.primary ? (
+    <Box marginRight="spacing.4" display="inline-flex">
+      <BaseButton onClick={actions.primary.onClick} intent={intent} contrast={contrast}>
+        {actions.primary.text}
+      </BaseButton>
+    </Box>
+  ) : null;
+  const secondaryActionParams: Nullable<Partial<SecondaryActionLinkButton>> = actions?.secondary
+    ? {
+        onClick: actions.secondary.onClick,
+      }
+    : null;
+  if (actions?.secondary && secondaryActionParams && 'href' in actions.secondary) {
+    // type guard with href to ensure this is now a SecondaryActionLinkButton
+    secondaryActionParams.href = actions.secondary.href;
+    secondaryActionParams.target = actions.secondary.target;
+    secondaryActionParams.rel = actions.secondary.rel;
+  }
+  const secondaryAction = actions?.secondary ? (
+    // Todo: Link font weight mismatch
+    <BaseLink contrast={contrast} intent={intent} {...secondaryActionParams}>
+      {actions.secondary.text}
+    </BaseLink>
+  ) : null;
+
+  const _actions =
+    primaryAction || secondaryAction ? (
+      <Box marginTop="spacing.3">
+        {primaryAction}
+        {secondaryAction}
+      </Box>
+    ) : null;
+
+  const onClickDismiss = (): void => {
+    if (onDismiss) {
+      onDismiss();
+    }
+    setIsVisible(false);
+  };
+  const closeButton = isDismissable ? (
+    <IconButton
+      accessibilityLabel="Dismiss alert"
+      onClick={onClickDismiss}
+      contrast={contrast}
+      size="large"
+      icon={CloseIcon}
+    />
+  ) : null;
+
+  const a11yProps = makeAccessible({
+    role: intent === 'negative' || intent === 'notice' ? 'alert' : 'status',
+    ...(intent === 'notice' && { liveRegion: 'polite' }), // override the implicit live region with role alert
+  });
+
+  if (!isVisible) {
+    return null;
+  }
+
+  return (
+    <StyledAlert
+      intent={intent}
+      contrastType={contrastType}
+      isFullWidth={isFullWidth}
+      isBorderless={isBorderless}
+      {...a11yProps}
+    >
+      {icon}
+      <Box flex={1} paddingLeft="spacing.3" paddingRight="spacing.3">
```

Comment:
What does `flex=1` do? We aren't even setting `display=flex`

Reply:
[`flex={1}`](https://stackoverflow.com/questions/37386244/what-does-flex-1-mean) is a shorthand. tl;dr - it tells this content to fill up all the available space in the container. Here, it makes the alert middle content to fill up the available space, while leaving only the needed space for icon (on left) and dismiss button (on right).

`display="flex"` isn't needed for flex child (it being a part of flex parent)